automerge-wasm/src/lib.rs

```diff
@@ -686,15 +739,100 @@ fn to_js_err<T: Display>(err: T) -> JsValue {
     js_sys::Error::new(&std::format!("{}", err)).into()
 }
 
-fn get(obj: &JsValue, prop: &str) -> Result<JS, JsValue> {
-    Ok(JS(Reflect::get(obj, &prop.into())?))
+fn get<J: Into<JsValue>>(obj: J, prop: &str) -> Result<JS, JsValue> {
+    Ok(JS(Reflect::get(&obj.into(), &prop.into())?))
 }
 
 fn set<V: Into<JsValue>>(obj: &JsValue, prop: &str, val: V) -> Result<bool, JsValue> {
     Reflect::set(obj, &prop.into(), &val.into())
 }
 
 struct JS(JsValue);
+struct AR(Array);
```

Comment:
This `JS` and `AR` stuff feels like it could be split into a separate file in this crate. I think it would make things a little more readable, especially as you could move all the `From`, `TryFrom`, `Into` stuff there as well.

Reply:
Thats a really good point.  The file is getting super cluttered.  Broke it up into 4.  Still not sure the right way to write Javascript iterop code but its evolving and I think its getting better.